File: Brute Force/Depth-First Search/tree.java
import org.algorithm_visualizer.*;

class Main {

    private static GraphTracer tracer = new GraphTracer();

    private static LogTracer logger = new LogTracer();

    private static Integer[][] G = {
        {0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
    };

    public static void main(String[] args) {
        tracer.log(logger);
        Layout.setRoot(new VerticalLayout(new Commander[]{tracer, logger}));
        tracer.set(G);
        tracer.layoutTree(0);
        Tracer.delay();

        DFS(0, null);
    }

    private static void DFS(int node, Integer parent) { // node = current node, parent = previous node
        tracer.visit(node, parent);
        Tracer.delay();
        for (int i = 0; i < G[node].length; i++) {
            if (G[node][i] != 0) { // if current node has the i-th node as a child
                DFS(i, node); // recursively call DFS
            }
        }
    }
}

File: Brute Force/Depth-First Search/tree.py
# import visualization libraries {
from algorithm_visualizer import GraphTracer, LogTracer, Tracer, Layout, VerticalLayout
# }

G = [  # G[i][j] indicates whether the path from the i-th node to the j-th node exists or not
    [0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
]

# define tracer variables {
tracer = GraphTracer()
logger = LogTracer()
tracer.log(logger)
Layout.setRoot(VerticalLayout([tracer, logger]))
tracer.set(G)
tracer.layoutTree(0)
Tracer.delay()
# }


def dfs(node, parent):  # node = current node, parent = previous node
    # visualize {
    tracer.visit(node, parent)
    Tracer.delay()
    # }
    for i in range(len(G[node])):
        if G[node][i]:  # if current node has the i-th node as a child
            dfs(i, node)  # recursively call DFS


dfs(0, None)
